### ISSUE TITLE
Slider widget get distort when rotate using rotated box

### ISSUE THREAD

**Issue by iliyami** [flutter/flutter#119448]
Description:
When I try to run the flutter web in the chrome with the Canvaskit renderer, there is no problem with rotating the slider widget, but when I try to do the same action in the chrome with the HTML renderer, the slider will be distorted in the rotated state.

1. Run the minimal reproduction project in the html renderer mode
2. Click the rotate button at the top right of the screen

**Expected results:** A vertical appropriate slider

**Actual results:** A vertical distorted slider

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

// This a test program for showing the Slider widget distortion i
ssue with the HTML renderer

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Slider Test',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Slider Test'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _sliderValue = 10;
  bool isRotated = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          children: [
            Text(
          'SliderValue: $_sliderValue',
        ),
            Align(
          alignment: Alignment.topRight,
          child: IconButton(
            onPressed: () {
              setState(() {
                isRotated = !isRotated;
              });
            },
            icon: const Icon(Icons.rotate_left),
          ),
        ),
            Expanded(
              child: RotatedBox(
                quarterTurns: (isRotated) ? 1 : 0,
                child: homeWidget(),
              ),
            ),
          ],
        ),
      ),
    );
  }

  Stack homeWidget() {
    return Stack(
      children: <Widget>[
        Slider(
          value: _sliderValue.toDouble(),
          onChanged: (value) {
            setState(() {
              _sliderValue = value.round();
            });
          },
          min: 0,
          max: 100,
        ),
      ],
    );
  }
}

```
</details>

<details>
  <summary>flutter doctor -v</summary>
  [✓] Flutter (Channel stable, 3.3.9, on macOS 13.0.1 22A400 darwin-x64, locale en-GB)
    • Flutter version 3.3.9 on channel stable at /Users/Parsa/Development/Frameworks/Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b8f7f1f986 (10 weeks ago), 2022-11-23 06:43:51 +0900
    • Engine revision 8f2221fbef
    • Dart version 2.18.5
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/Parsa/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.74.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.56.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

</details>

When the slider is in horizontal mode:
![image_2023-01-29_140933020](https://user-images.githubusercontent.com/18049527/215320686-be3b5938-3afc-41fa-9c78-10c65c029fc1.png)

When the slider is in the vertical mode using rotated box:
<img width="496" alt="image" src="https://user-images.githubusercontent.com/18049527/215320710-2d2f1f49-30f1-428c-91bf-cc2c94140581.png">

cc @TahaTesser

**Comment by exaby73** [comment]
## Triage report

I can reproduce this issue with `Master (3.7.0-29.0.pre.26)`. This only occurs on the HTML renderer on web. Canvaskit, desktop and mobile work fine in my testing

<details>
<summary>Simplified Code Sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Slider Test',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(title: 'Slider Test'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool isRotated = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: RotatedBox(
          quarterTurns: (isRotated) ? 1 : 0,
          child: const HomeWidget(),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () => setState(() => isRotated = !isRotated),
        child: const Icon(Icons.rotate_left),
      ),
    );
  }
}

class HomeWidget extends StatefulWidget {
  const HomeWidget({super.key});

  @override
  State<HomeWidget> createState() => _HomeWidgetState();
}

class _HomeWidgetState extends State<HomeWidget> {
  int _sliderValue = 10;

  @override
  Widget build(BuildContext context) {
    return Stack(
      children: <Widget>[
        Slider(
          value: _sliderValue.toDouble(),
          onChanged: (value) {
            setState(() {
              _sliderValue = value.round();
            });
          },
          min: 0,
          max: 100,
        ),
      ],
    );
  }
}
```

</details>

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/50555895/215413433-4bcfa3ab-fc85-4467-b219-cce76fa87987.mov

</details>

**Comment by iliyami** [comment]
Any news for this issue ?
@darshankawar @exaby73

**Comment by exaby73** [comment]
@iliyami Please refrain from mentioning people at random. The issue is being looked. It's an engine issue, and take a bit of time to debug and investigate

**Comment by iliyami** [comment]
> @iliyami Please refrain from mentioning people at random. The issue is being looked. It's an engine issue, and take a bit of time to debug and investigate

Yes, I know. I just wanted to check if the issue is being looked or not. Thanks.

**Comment by bleroux** [comment]
I have investigated this further. I don't have a fix for it but I can provide some insights.

--
**First step** was to simplify the code sample in order to keep the widget tree as small as possible: 
- It eliminates possible causes (for instance, the initial code sample uses a `Stack`).
- When investigating HTML renderer issues, one can rely on Chrome Dev Tools to inspect the produced DOM. The less widgets the app contains the less complex is the DOM.

<details>
<summary>Simplified code sample</summary>

```dart
import 'package:flutter/material.dart';

// This a test program for showing the Slider widget distortion i

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return Directionality(
      textDirection: TextDirection.ltr,
      child: Material(
        child: RotatedBox(
          quarterTurns: 1,
          child: Slider(
            value: 50.0,
            onChanged: (_) {},
            min: 0,
            max: 100,
          ),
        ),
      ),
    );
  }
}
```
</details>

--
**Second step** was to find if it can be reproduced when replacing the `Slider` with simple or complex widgets.
Strangely, this can not be reproduced using various widgets I tested:
- a simple one: `Text`.
- a more complex one: `ElevatedButton` (with splash effect).
- a close one: `CupertinoSlider`.

--
**Third step** was to make changes in `Slider` code to find which change can eliminate the rendering issue.
This led me to removing the `CompositedTransformTarget` use here:
https://github.com/flutter/flutter/blob/3af30ff5913f6a04a270bf3d2167cf4babbad1c4/packages/flutter/lib/src/material/slider.dart#L873
Without this additional layer, the Slider is rendering correctly.

--
**Next step** is to try to repro by replacing the Slider widget with a simpler one (to isolate which rendering combinaison produces this error). For the moment, I have not found one (tested with a `CustomPainter` and with a custom widget with a custom `RenderObject`).
